### ISSUE TITLE
[Bug]: azure_ad_token not processed correctly as an argument

### ISSUE THREAD

**Issue by vilmar-hillow** [BerriAI/litellm#1502]
### What happened?

`azure_ad_token` has no effect when passed as an argument to `litellm.completion`.
`litellm.completion(..., azure_ad_token=token, ...)` call fails with:
```
litellm.exceptions.APIError: AzureException - Missing credentials. Please pass one of `api_key`, `azure_ad_token`, `azure_ad_token_provider`, or the `AZURE_OPENAI_API_KEY` or `AZURE_OPENAI_AD_TOKEN` environment variables.
```

However, if the same `token` is set as env variable (e.g. via `os.environ["AZURE_AD_TOKEN"] = token`), the call is successful.

Debug has some hints: when passed as an argument, the token is sent as `extra_body` in a payload. When set via env variable, it's sent correctly in a header.

**Comment by ghost** [comment]
just pushed the fix - https://github.com/BerriAI/litellm/commit/e0aaa94f28304e362dabd39b164940834ac0fa50


should be live soon in v`1.18.3`. Also fixed the relevant test, to prevent future regressions.

**Comment by ghost** [comment]
Please re-open the issue if the error persists

**Comment by ishaan-jaff** [comment]
Hi @vilmar-hillow  any chance we can hop on a call to better understand how we can improve litellm for you ?

Link to my cal for your convenience
https://calendly.com/d/4mp-gd3-k5k/litellm-1-1-onboarding-chat?month=2024-02

**Comment by vilmar-hillow** [comment]
@ishaan-jaff At the moment I'm experimenting with several LLMs, and litellm is useful for having a standard API across different model types. I also see some of the potentially useful cool features for the future uses (proxy, cost tracking). I'll make sure to set up a call once I have more feedback on what would be useful, and report any issues or contribute to the codebase as I'm exploring

**Comment by vilmar-hillow** [comment]
Not sure if related, but `litellm.embedding(..., azure_ad_token=token, ...)` fails with:

```
APIError: AzureException - Embeddings.create() got an unexpected keyword argument 'azure_ad_token'
```

@krrishdholakia Let me know if you need more details or want me to open a new issue!
Opened a PR with potential fix: #2142
